platform/linux-generic/odp_schedule_scalable.c

```diff
@@ -1025,9 +1027,25 @@ static void schedule_order_unlock(unsigned lock_index)
 	atomic_store_release(&rctx->rwin->olock[lock_index],
 			     rctx->sn + 1,
 			     /*readonly=*/false);
+	atomic_store_release(&rctx->rwin->lock_acquired,
+			     rctx->rwin->lock_count + 1, false);
 	rctx->olock_flags |= 1U << lock_index;
 }
 
+static void schedule_order_unlock_lock(unsigned lock_index)
+{
+	struct reorder_context *rctx;
+
+	rctx = sched_ts->rctx;
+	if (odp_unlikely(rctx == NULL || rctx->rwin == NULL)) {
+		ODP_ERR("Invalid call to odp_schedule_order_unlock_lock\n");
+		return;
+	}
+	schedule_order_unlock(__atomic_load_n(&rctx->rwin->lock_acquired,
+					      __ATOMIC_ACQUIRE));
```

Comment:
A simple reference to the stored `lock_index` is sufficient here. No need for atomics.